### ISSUE TITLE
N:1 tag matching

### ISSUE THREAD

**Issue by matkoniecz** [ENT8R/tags2names#2]
I am not sure is it feasible but for some uses it may be desirable to control how specific description is supposed to be.

For example in one of my programs I require something like

>   settlements = ['country', 'state', 'province', 'region',
        'district', 'county', 'municipality', 'city', 'borough', 'suburb',
        'quarter', 'neighbourhood', 'city_block', 'town', 'village',
        'hamlet', 'isolated_dwelling', 'farm']
>  return "settlement" if settlements.include?(element["tags"]["place"])
>  return "shop" if element["tags"]["shop"] != nil and element["tags"]["shop"] != "no"

If getting this type of less precise description is feasible in this repo it would be very nice to avoid duplication of effort.

**Comment by matkoniecz** [comment]
One hacky solution is to have a separate en-lesss-detailed-name language but it is extremely hacky.

**Comment by ENT8R** [comment]
> it would be very nice to avoid duplication of effort.

Yes, indeed, this is always nice 👍 

But if this kind of filtering would be done within this project, you would need to decide to use one particular programming language, which is maybe not applicable to other projects.

Consider the following use case:
The OSM iD editor uses mostly javascript. If the code of this dictionary would be written in e.g. Python, they would need to use this language too...
Basically, my point is, that I don't want other projects to use the language that this project uses...

**Comment by matkoniecz** [comment]
It should be possible to express it as data, 

it may be overly complex

```
settlement:
    - place=village
    - place=city
    - place=hamlet
    
shop:
    - [shop=*,shop!=no]
```

but it should be possible to express it as

```
shop:
  alcohol: shop
  baby_goods: shop
  bag: shop
  bakery: shop
```

**Comment by ENT8R** [comment]
Mmh... But using your second variant, you can also specify a generic name for all OSM tags like:
```yml
generic:
  shop: shop
  leisure: free time facility
```
Which is IMO the wrong way to create a dictionary which should cover all OSM tags...
If it is possible to get a more precise name of a tag, why don't do it?

**Comment by matkoniecz** [comment]
> If it is possible to get a more precise name of a tag, why don't do it?

In my case - I want to classify tags of some group of objects and get report "1717 settlements, 1621 shops, 172 businesses".

But now that I type this I see that classification would strongly depend on situation and making something generic fitting everybody is not going to happen.
